site/documentation/content/api/telemetry/index.md

```diff
@@ -118,3 +121,134 @@ The *Business Application* can only consume telemetry messages that have been up
+This guarantees that the message will not be lost as long as at least one in-sync replica remains alive.
+
+If *retries* are enabled and *max.in.flight.requests.per.connection* is set to a value greater than `1`, the order of the messages is no longer guaranteed by Kafka. 
+The only exception is the *idempotent* producer, which can handle *max.in.flight.requests.per.connection* to be up to `5` with retries enabled while still maintaining the message order (since Kafka version 1.0.0).
+
+{{% /note %}}
+
+**Preconditions**
+
+1. Either the topic `telemetry-${tenant_id}` exists, or the broker is configured to automatically create topics on demand.
+1. The client is authorized to write to the topic.
+1. The device for which the adapter wants to send telemetry data has been registered (see [Device Registration API]({{< relref "/api/device-registration" >}})).
+
+**Message Flow**
+
+It is up to the discretion of the protocol adapter whether it wants to use *AT LEAST ONCE* or *AT MOST ONCE* delivery semantics.
+To achieve *AT LEAST ONCE* delivery semantics the protocol adapter MUST request acknowledgements from *all* in-sync-replicas in the Kafka cluster.
+For *AT MOST ONCE* delivery semantics the protocol adapter acknowledges the message to the device without waiting for any acknowledgements from the Kafka cluster.
+If *retries* is greater than zero, *max.in.flight.requests.per.connection* MUST be set to `1`, except if *enable.idempotence* is set to `true`.
+
+{{% note title="TO BE DISCUSSED" %}}
+**_Should we add a table similar to the one in the AMQP Telemetry API? It could look like this:_**
+
+| acks  | retries   |  max.in.flight.requests.per.connection    | enable.idempotence    | Delivery semantics |
+| :---: | :-----:   | :-------------------------------------:   | :-------------------: | :----------------- |
+| 0     | 0  | >0 | `false` | Setting *acks* to `0` allows for adapters to implement *AT MOST ONCE* delivery semantics only. This is the fastest mode of delivery but has the drawback of potential loss of messages without notice. |
+| 1     | 0  | >0 | `false` | Setting *acks* to `1` allows for adapters to implement *AT MOST ONCE* delivery semantics only. If the leader goes down before the followers finished replication, then the message will be lost. | 
+| 1     | >0 | 1  | `false` | Setting *acks* to `1` allows for adapters to implement *AT MOST ONCE* delivery semantics only. When retries are enabled, *max.in.flight.requests.per.connection* MUST be set to `1`. | 
+| all   | 0  | >0 | `false` | Setting *acks* to `all` allows for adapters to implement both *AT LEAST ONCE* or *AT MOST ONCE* delivery semantics, depending on whether the adapter waits for and considers the acknowledgement it receives from the Kafka cluster or not. Disabling retries might cause messages to fail in case of high load or transient transmission failures. |
```

Comment:
How about using `either` here? I think when you say `both`, then it means both *AT LEAST ONCE* and *AT MOST ONCE*. 

maybe it is worth to distinguish how this implements AT LEAST ONCE or AT MOST ONCE delivery semantics. Do you mean that it implements `AT LEAST ONCE` when there is at least one alive in-sync replica? And it implements `AT MOST ONCE` when no in-sync replica is  alive?

Reply:
@Alfusainey Thank you for your feedback. Please have a look at my last change if it resolves this issue.